extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -338,6 +339,46 @@ public void create(UUID jobId, String key, InputStream stream) {
     googleTempFileStore.create(jobId, key, stream);
   }
 
+  @Override
+  public void addCounts(UUID jobId, Map<String, Integer> newCounts) throws IOException {
+    if (newCounts == null) {
+      return;
+    }
+
+    Transaction transaction = datastore.newTransaction();
+
+    for (String dataType : newCounts.keySet()) {
+      Key key = getCountsKey(jobId, dataType);
+      Entity current = datastore.get(key);
+      Integer oldCount = 0;
+
+      if (current != null && current.getNames().contains(COUNTS_FIELD)) {
+        oldCount = Math.toIntExact(current.getLong(COUNTS_FIELD));
```

Comment:
Is this cause Datastore only allows longs? If so, let's add a comment.

Reply:
Done